lib/api.js

```diff
@@ -0,0 +1,275 @@
+      var version = browser[ versionField ],
+        browserId = getBrowserId( browser );
+
+      // ignore devices that don't have versions
+      if ( !version ) {
+        return;
+      }
+
+      // ignore pre-release versions
+      if ( /\s/.test( version ) ) {
+        return;
+      }
+
+      if ( parseFloat( version ) >
+          (parseFloat( latest[ browserId ] ) || 0) ) {
+        latest[ browserId ] = version;
+      }
+    });
+  }
+});
+
+
+// Versions
+var apiVersions = {
+
+  1: {
+    useHttp: true,
+
+    versionField: "version",
+
+    _getBrowsers: function( fn ) {
+      this.request({
+        path: this.path( "/browsers" )
+      }, fn );
+    },
+
+    getBrowserId: function( browser ) {
+      return browser.browser;
+    }
+  },
+
+  2: {
+    useHttp: true,
+
+    versionField: "version",
+
+    _getBrowsers: function( fn ) {
+      this.request({
+        path: this.path( "/browsers" )
+      }, function( error, osBrowsers ) {
+        if ( error ) {
+          return fn( error );
+        }
+
+        fn( null, [].concat.apply( [],
+          Object.keys( osBrowsers ).map(function( os ) {
+            return osBrowsers[ os ].map(function( browser ) {
+              browser.os = os;
+              return browser;
+            });
+          })
+        ));
+      });
+    },
+
+    getBrowserId: function( browser ) {
+      return browser.os + ":" + (browser.browser || browser.device);
+    }
+  },
+
+  3: {
+    useHttp: true,
+
+    versionField: "browser_version",
+
+    _getBrowsers: function( fn ) {
+      this.request({
+        path: this.path( "/browsers?flat=true" )
+      }, fn );
+    },
+
+    getBrowserId: function( browser ) {
+      var id = browser.os + ":" + browser.os_version + ":" + browser.browser;
+      if ( browser.device ) {
+        id += ":" + browser.device;
+      }
+
+      return id;
+    },
+
+    getApiStatus: function( fn ) {
+      this.request({
+        path: this.path( "/status" )
+      }, fn );
+    }
+  },
+
+  4: {
+    versionField: "browser_version",
+
+    _getBrowsers: function( fn ) {
+      this.request({
+        path: this.path( "/browsers?flat=true" )
+      }, fn );
+    },
+
+    getBrowserId: function( browser ) {
+      var id = browser.os + ":" + browser.os_version + ":" + browser.browser;
+      if ( browser.device ) {
+        id += ":" + browser.device;
+      }
+
+      return id;
+    },
+
+    getApiStatus: function( fn ) {
+      this.request({
+        path: this.path( "/status" )
+      }, fn );
+    }
+
+  }
+
+}
+
+
+module.exports = {
+  createClient: function ( settings ) {
+
+    // get version-specific properties
+    var version = settings.version || Math.max.apply(null, Object.keys(apiVersions));
+    var versionSettings = apiVersions[ version ];
+    if ( !versionSettings ) {
+      throw new Error( "Invalid version" );
+    }
+    apiClient.prototype.version = version;
+    extend( apiClient.prototype, versionSettings);
```

Comment:
This is a destructive operation. Every time a new client is created, you're changing the base prototype. This means that two clients for different versions will step on top of each other and whichever is created first will break. Please don't change the way that the versioned APIs are handled.